bootstrapper/internal/initserver/initserver.go

```diff
@@ -187,12 +205,83 @@ func (s *Server) Init(ctx context.Context, req *initproto.InitRequest) (*initpro
 	}
 
 	log.Infof("Init succeeded")
+
 	return &initproto.InitResponse{
 		Kubeconfig: kubeconfig,
 		ClusterId:  clusterID,
 	}, nil
 }
 
+// GetLogs gets and streams the requested systemd logs.
+func (s *Server) GetLogs(req *initproto.LogRequest, stream initproto.API_GetLogsServer) error {
+	s.shutdownLock.TryRLock() // try to lock the process if a call comes in
+	log := s.log.With(zap.String("peer", grpclog.PeerAddrFromContext(stream.Context())))
+	log.Infof("GetLogs called")
+
+	log.Infof("Validating the init secret...")
+	if err := bcrypt.CompareHashAndPassword(s.initSecretHash, req.InitSecret); err != nil {
+		return status.Errorf(codes.Internal, "invalid init secret: %s", err)
+	}
+
+	logPipe, err := s.journaldCollector.Start()
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed starting the log collector: %s", err)
+	}
+
+	// decode the master secret
+	cfg, err := uri.DecodeMasterSecretFromURI(s.kmsURI)
```

Comment:
I don't think we should assume that the `kmsURI` contains a master secret, since it could also contain various different kms endpoints. We either need to re-think the crypto or we need to setup the KMS as done in `Init()` in line 146.

Reply:
I don't see a way in init which derives a key that the client could generate as well. We could use the init secret for deriving a key instead of the master secret, or use asymetric cryptography for a key exchange, at least those are the only two options that came to my mind. What do you think would work best/do you have a different idea which could work?